city_scrapers/spiders/il_port_district.py

```diff
@@ -0,0 +1,181 @@
+        meeting_time = "9:00am"
+
+        self._validate_location(response)
+
+        for i, row in enumerate(strong_meetings + rows[1:]):
+            if i >= len(strong_meetings):
+                meetings_dates = row.xpath(".//div/text()").extract()
+                if not meetings_dates:
+                    continue
+            else:
+                meetings_dates = row
+
+            for i, date in enumerate(meetings_dates):
+                if not date:
+                    continue
+                special_meeting = None
+                if date.startswith("***") or date.endswith("***"):
+                    meeting_time = changed_meeting_time
+                elif date.startswith("**") or date.endswith("**"):
+                    special_meeting = "Special "
+                elif date.startswith("*") or date.endswith("*"):
+                    new_date = re.findall(r"(?<=\()(.*?)(?=NEW)", date)
+                    date = new_date[0][:-3] if new_date else date
+
+                date = date.strip(" *")
+
+                title = meeting_types[i] if not special_meeting\
+                    else special_meeting + meeting_types[i]
+
+                classification = self._parse_classification(i, meeting_types[i])
+
+                start = self._parse_start(year, date, meeting_time)
+
+                agendas_links = self._parse_agendas_links(classification)
+
+                minutes_links = None
+
+                if classification == "Board":
+                    minutes_links = self._parse_minutes_links(start)
+                links = agendas_links if not minutes_links else\
+                    agendas_links + minutes_links
+
+                meeting = Meeting(
+                    title=title,
+                    description="",
+                    classification=classification,
+                    start=start,
+                    end=None,
+                    all_day=False,
+                    time_notes="",
+                    location=self.location,
+                    links=links,
+                    source=response.url
+                )
+
+                meeting["status"] = self._get_status(meeting)
+                meeting["id"] = self._get_id(meeting)
+
+                yield meeting
+
+    def parse_agendas(self, response):
+        file_prefix = response.xpath("//tr/td[@class='views-field views-field-title']/text()")\
+            .extract()
+        file_prefix = [x.strip("\n ") for x in file_prefix]
+        file_links = response.xpath("//tr/td/a[@class='file-download']/@href").extract()
+        name_pattern = r"(?<=documents\/)(.*?)(?=.pdf)"
+        file_names = [re.findall(name_pattern, x)[0].replace('%', ' ') for x in file_links]
+
+        self.files_list = []
+
+        for link, prefix, name in list(zip(file_links, file_prefix, file_names)):
+            self.files_list.append({'title': prefix + " " + name, 'href': link})
```

Comment:
If we know these files are always agendas the title can just be "Agenda"

Reply:
I should use "Agenda" title for each file, or distinguish between different Agenda types (Board, Committee) an associate them to corresponding meetings?

Because if I use "Agenda" for each file name then this function will not work properly, as it looks for the "Board" or "Committee" phrases in the file names:
```
    def _parse_agendas_links(self, classification):
        if classification == 'Board':
            link_list = [d for d in self.files_list if 'Board' in d['title']]
        elif classification == 'Committee':
            link_list = [d for d in self.files_list if 'Committee' in d['title']]
        else:
            link_list = []

        return link_list
```
How about if I get rid of the file name part that was derived from the url, so instead of the following:
- Leases and Agreement Committee Agenda L 26A 20Agenda 20November 202019
We will have:
- Leases and Agreement Committee Agenda ?

Reply:
Gotcha, yeah that should work and at least leave the values a bit cleaner